src/Features/CSharp/Portable/GoToDefinition/CSharpGoToDefinitionSymbolService.cs

```diff
@@ -20,5 +21,90 @@ public CSharpGoToDefinitionSymbolService()
 
         protected override ISymbol FindRelatedExplicitlyDeclaredSymbol(ISymbol symbol, Compilation compilation)
             => symbol;
+
+        protected override int? GetTargetPositionIfControlFlow(SyntaxToken token)
+        {
+            var node = token.Parent;
+            if (node is null)
+            {
+                return null;
+            }
+
+            switch (token.Kind())
+            {
+                case SyntaxKind.ContinueKeyword:
+                    var foundContinuedLoop = TryFindContainingLoop(node);
+
+                    return foundContinuedLoop?.IsContinuableConstruct() == true
+                        ? foundContinuedLoop.GetFirstToken().Span.Start
+                        : null;
+
+                case SyntaxKind.BreakKeyword:
+                    var foundBrokenLoop = TryFindContainingLoop(node);
+
+                    return foundBrokenLoop?.IsBreakableConstruct() == true
+                        ? foundBrokenLoop.GetLastToken().Span.End
+                        : null;
+
+                case SyntaxKind.YieldKeyword:
+                    var nextToken = token.GetNextToken();
+                    switch (nextToken.Kind())
+                    {
+                        case SyntaxKind.ReturnKeyword:
+                            return GetTargetPositionIfControlFlow(nextToken);
```

Comment:
so this is weird.  if we have `yield return` the nwe recurse into thsi method, which will hit the `return` case below.   But the `return` case exactly matches teh `yield break` code.  So all 3 of yield-return/yield-break/return seem to have identical logic, just written differently.

Reply:
Thanks. I've consolidated the handling of yield-return/yield-break/return. I had a bug with `yield break$$` scenario.